### ISSUE TITLE
assumptions stumble with this expression for zero

### ISSUE THREAD

**Issue by smichr** [sympy/sympy#14558]
```
>>> z = S('''sqrt(3)*(-2**(2/3)*3**(1/3)*I + 12**(1/3)*I)''')
>>> z._eval_is_imaginary()
False
>>> z._eval_is_zero()
False
>>> z.is_rational
False
```

**Comment by smichr** [comment]
This is a helpful routine to compare what the object's evaluation are relative to the deduced values:
```
def compare(z):
    print('\t_eval\tis_')
    for k in sorted([w for w in dir(z) if 'eval_is' in w]):
        try:
            a = getattr(z, k)(), getattr(z, k.lstrip('_eval_'))
            args = a + (k.split('_')[-1], )
            if a[0] != a[1] and None not in a:
                print("%-->\ts\t%s\t%s" % args)
            elif a[0] != a[1]:
                print(">\t%s\t%s\t%s" % args)
            else:
                print("\t%s\t%s\t%s" % args)
        except:
            pass

>>> compare(sqrt(2))
        _eval   is_
        True    True    algebraic
        True    True    complex
>       None    False   composite
>       None    False   even
        True    True    finite
        False   False   imaginary
        False   False   integer
        False   False   negative
>       None    False   odd
        None    None    polar
        True    True    positive
>       None    False   prime
        False   False   rational
        True    True    real
        False   False   zero
```

**Comment by smichr** [comment]
merged into #8046
